CMSgov/mint-app
# [EASI-4147] generic audit translation work from first POC

# EASI-4147


## Changes and Description

- Pulling in Changes from POC branch
    - Note, this structure is going to be changed, putting up a PR to get some changes in the Feature Branch




## How to test this change
1. run  `go run ./cmd/dbseed/*go`
2. Select seed, press enter
3. Ensure DB is seeded.
4. Unselect seed, and select translate
5. Ensure translations audit entries appear in the DB.

```gql
query translatedAuditChangeCollection {
  translatedAuditChangeCollection(modelPlanID: "{{modelPlanID}}") {
    id
    modelName
    tableID
    tableName
    primaryKey
    date
    action

    fieldName
    fieldNameTranslated
    old
    oldTranslated
    new
    newTranslated
    actorID
    actorName

    changeID
    metaData {
    ... on TranslatedAuditMetaBaseStruct{
            version
            tableName
        }
    }

    createdBy
    # createdByUserAccount
    createdDts
    modifiedBy
    # modifiedByUserAccount
    modifiedDts
  }
}

```



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
